src/Security.php

```diff
@@ -28,13 +28,13 @@ protected static function heuristicScan($xml)
      * Scan XML string for potential XXE and XEE attacks
      *
      * @param   string $xml
-     * @param   DomDocument $dom
+     * @param   DomDocument|null $dom
      * @param   int $libXmlConstants additional libxml constants to pass in
      * @param   callable $callback the callback to use to create the dom element
      * @throws  Exception\RuntimeException
      * @return  SimpleXMLElement|DomDocument|boolean
      */
-    private static function scanString($xml, DOMDocument $dom = null, $libXmlConstants, callable $callback)
+    private static function scanString($xml, DOMDocument|null $dom = null, $libXmlConstants, callable $callback)
```

Comment:
```suggestion
    private static function scanString($xml, DOMDocument|null $dom, $libXmlConstants, callable $callback)
```

This is a private method - you can do anything you like to it!